WordPress/Classes/ViewRelated/Portfolio/PortfolioListViewController.swift

```diff
@@ -0,0 +1,514 @@
+            let searchPredicate = NSPredicate(format: "postTitle CONTAINS[cd] %@", searchText)
+            predicates.append(searchPredicate)
+        }
+
+        let predicate = NSCompoundPredicate(andPredicateWithSubpredicates: predicates)
+        return predicate
+    }
+
+    // MARK: - Table View Handling
+
+    func sectionNameKeyPath() -> String {
+        let sortField = filterSettings.currentPostListFilter().sortField
+        return Project.sectionIdentifier(dateKeyPath: sortField.keyPath)
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return type(of: self).portfolioSectionHeaderHeight
+    }
+
+    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        if section == tableView.numberOfSections - 1 {
+            return WPDeviceIdentification.isRetina() ? 0.5 : 1.0
+        }
+        return 0.0
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView! {
+        let sectionInfo = tableViewHandler.resultsController.sections?[section]
+        let nibName = String(describing: PageListSectionHeaderView.self)
+        let headerView = Bundle.main.loadNibNamed(nibName, owner: nil, options: nil)![0] as! PageListSectionHeaderView
+
+        if let sectionInfo = sectionInfo {
+            headerView.setTite(sectionInfo.name)
+        }
+
+        return headerView
+    }
+
+    func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView! {
+        if section == tableView.numberOfSections - 1 {
+            return sectionFooterSeparatorView
+        }
+        return UIView(frame: CGRect.zero)
+    }
+
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+
+        let project = projectAtIndexPath(indexPath)
+
+        if project.remoteStatus != .pushing && project.status != .trash {
+            editProject(project)
+        }
+    }
+
+    @objc func tableView(_ tableView: UITableView, cellForRowAtIndexPath indexPath: IndexPath) -> UITableViewCell {
+        if let windowlessCell = dequeCellForWindowlessLoadingIfNeeded(tableView) {
+            return windowlessCell
+        }
+
+        let project = projectAtIndexPath(indexPath)
+
+        let identifier = cellIdentifierForProject(project)
+        let cell = tableView.dequeueReusableCell(withIdentifier: identifier, for: indexPath)
+
+        configureCell(cell, at: indexPath)
+
+        return cell
+    }
+
+    override func configureCell(_ cell: UITableViewCell, at indexPath: IndexPath) {
+
+        guard let cell = cell as? BasePageListCell else {
+            preconditionFailure("The cell should be of class \(String(describing: BasePageListCell.self))")
+        }
+
+        cell.accessoryType = .none
+
+        if cell.reuseIdentifier == type(of: self).projectCellIdentifier {
```

Comment:
You can replace the if with a switch on this one (which would improve readability).

Alternatively (not sure if it's worth it, since there are just two cells), but we could do something like:

```
enum ReuseIdentifier: String {
    case project
    case restore
}
```

That way, this whole snippet could look like this:

guard let identifier = ReuseIdentifier(rawValue: cell.reuseIdentifier) else {
   return
}
switch identifier {
case .project:
    xxxx
case .restore:
   yyyy
}
```

Still, since it's just two cell types, not sure if it's worth it!